README.md
# hello-nextflow
A minimal DSL2 nextflow example for processing fastq read pairs

## Setup

Install scripts for APHA bioinformatics apprentice project.

The scripts within this repository will automatically download and install the required software for
the APHA bioinformatics project. These are:

- [Nextflow](https://github.com/nextflow-io/nextflow)
- [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
- [fastp](https://github.com/OpenGene/fastp)
- [Trimmomatic](https://github.com/usadellab/Trimmomatic)
- [Unicycler](https://github.com/rrwick/Unicycler)
- [prokka](https://github.com/tseemann/prokka)
- [MultiQC](https://github.com/ewels/MultiQC)

along with all of their depedencies.

### Installation

To dowload and run the install script, execute the following commands from within your user home
directory (e.g. `/home/nickpestell`):

1. `sudo apt-get update`
2. `sudo apt-get install git`
3. `git clone https://github.com/APHA-CSU/hello-nextflow.git`
4. `cd apprentice-install`
5. `bash install.bash`

If ran with no errors the final output should be:

```
Installation complete!
Apprentices, YOU ROCK!
```

## Run

To run:
```
nextflow run hello.nf
```


.nextflow/history
2022-05-03 11:07:30	-	mighty_yonath	-	56503691ebd13d4b795bdaa317807c02	9d21f779-b4ca-408c-8dcc-f8b1dbf5d67b	nextflow run hello.nf
2022-05-03 11:11:15	3s	drunk_bardeen	OK	a28d45b7c2d4b7388172578d4d738f5b	bc5da9cb-ea88-43ce-bd2e-a3f8b7815c23	nextflow run hello.nf
2022-05-03 11:11:53	2.4s	confident_boyd	OK	efc75db0cab66f386096f38bc0c2a769	e5a410f1-f60a-4ee5-8158-fce5f1ead2dc	nextflow run hello.nf
2022-05-03 11:15:05	1.9s	extravagant_saha	OK	99d9d18ef1bba35376c91e0c4b339aab	9be14e1e-8477-4b36-82c4-9c63a2a7051c	nextflow run hello.nf
2022-05-03 11:15:27	2.3s	zen_lorenz	OK	99d9d18ef1bba35376c91e0c4b339aab	23039e47-3a3a-435a-8d9c-ad47b492c1e8	nextflow run hello.nf
2022-05-03 11:16:47	2.1s	cheeky_bell	OK	3632c8e3364ede473e7644872b923a83	1da71041-231e-4ee8-a440-45351209aa37	nextflow run hello.nf
2022-05-03 11:21:13	2.9s	mad_albattani	OK	185d87a613cc808999c256b81c7d15ae	5f4fd2f1-134c-4270-aa2e-5f7842a6151c	nextflow run hello.nf
2022-05-03 11:21:45	2.3s	astonishing_hopper	OK	185d87a613cc808999c256b81c7d15ae	44c34357-8ff8-4cad-90dd-2fbe5017928f	nextflow run hello.nf
2022-05-03 11:24:05	-	voluminous_shannon	-	a3a693cc293853e12dc50595098f9f38	6f4e81bb-96da-47fa-8543-49a3c898eed9	nextflow run hello.nf
2022-05-03 11:24:52	-	disturbed_pike	-	319461fa873cfdca314c6a3f2e5c7b6d	c264440b-b93d-47f2-ab07-4405cc098824	nextflow run hello.nf
2022-05-03 11:25:22	2.4s	awesome_raman	OK	2512951406e19e5d8ee5cc4b64e93c3d	e2155db8-2b70-4f07-b035-533d0bd28368	nextflow run hello.nf
2022-05-03 11:27:39	2.5s	cranky_poitras	OK	6f6e09c5802e61595bfad511b1fab59a	26185961-1315-44b8-ab3b-ab380c203bb2	nextflow run hello.nf
2022-05-03 11:29:16	2.4s	nostalgic_lichterman	OK	0b9771c2c682b17d7e7053c0fdc55b39	68a6d01c-ccef-4aa6-9dbc-bff890c783e0	nextflow run hello.nf
2022-05-03 11:30:15	3s	gigantic_visvesvaraya	OK	98dbca7b0f633110357e54c5594c905f	8b6e08b4-a74f-4241-9b7d-f17f9bcd1823	nextflow run hello.nf
2022-05-03 11:33:26	-	clever_ritchie	-	949853295b634e3757a0c22293cc7926	6b4d7492-ba7f-4085-915a-c4a9ccf9f7ff	nextflow run hello.nf
2022-05-03 11:33:38	-	big_faggin	-	07cefb30bdc1889dd4c925e528f0839f	1d073c72-8530-43b2-bac2-0cb0a3926046	nextflow run hello.nf
2022-05-03 11:33:58	-	chaotic_darwin	-	3ea6754a01cdcfbf0d78f82533249756	4123fcb0-b803-458e-b0eb-3dfa97246a3f	nextflow run hello.nf
2022-05-03 11:34:29	3.1s	thirsty_lamport	ERR	2238521d6a8f36e39764295cc7c5fdfe	7bdeebf2-fd21-4d97-91ce-64b1d46e2429	nextflow run hello.nf
2022-05-03 11:34:53	2.7s	nostalgic_shockley	OK	82e528da80f50419c0b641f0a154bb61	557717e5-de78-4cd3-b5eb-349b8787c32b	nextflow run hello.nf


samples/toms.sh
#!/bin/bash

# Set global parameters (edit these parameters)
threads=4
baseDir=$PWD
read1=SRR11947553_sub_1.fq.gz
read2=SRR11947553_sub_2.fq.gz

# Create new directory to store all results
mkdir $baseDir/bash_results
resultsDir=$baseDir/bash_results

# Create new directories for each task
mkdir $resultsDir/1.fastqc_results
mkdir $resultsDir/2.clean_reads
mkdir $resultsDir/3.genome_assembly
mkdir $resultsDir/4.annotate_genome

# Define paths to directories
fastqcDir=$resultsDir/1.fastqc_results
cleanDir=$resultsDir/2.clean_reads
assemblyDir=$resultsDir/3.genome_assembly
annotateDir=$resultsDir/4.annotate_genome

# To run this script, navigate to the baseDir and type this command:
# bash run_bash.sh
# Note: you need to name this file run_bash.sh


# -------------------
# 1. Run FastQC and MultiQC
#--------------------

# Execute FastQC
fastqc --outdir $fastqcDir --threads ${threads} $baseDir/${read1} $baseDir/${read2} 

# Execute MultiQC
multiqc $fastqcDir --outdir $fastqcDir


# -------------------
# 2. Clean reads using fastp
#--------------------

# Execute fastp
fastp -i $baseDir/${read1} -I $baseDir/${read2} -o $cleanDir/${read1} -O $cleanDir/${read2} -j $cleanDir/fastp.json -h $cleanDir/fastp.html -q 30 --trim_poly_g --length_required 80 --thread ${threads}


# -------------------
# 3. Assemble clean reads using Unicycler
#--------------------

# Execute Unicycler
unicycler -1 $cleanDir/${read1} -2 $cleanDir/${read2} -o $assemblyDir --threads ${threads}


# -------------------
# 4. Annotate assembly using Prokka
#--------------------

# Execute Prokka
# conda activate prokka
prokka --outdir $annotateDir --force --prefix ${read1%_sub*} $assemblyDir/assembly.fasta




install/install.bash
#!/usr/bin/bash

set -e

################## ARGS ##############################

BIOTOOLS_PATH=~/biotools/

sudo apt-get -y update && sudo DEBIAN_FRONTEND=noninteractive apt-get install -y \
	unzip \
	default-jre \
	git \
	wget \
	gcc \
	g++ \
	make \
	libdatetime-perl \
	libxml-simple-perl \
	libdigest-md5-perl \
	bioperl \
	python3 \
	python3-pip \
	ncbi-blast+ \
	ncbi-tools-bin

echo yes | cpan
sudo cpan Bio::Perl
sudo cpan Bio::SearchIO::hmmer3

#python
pip3 install setuptools multiqc
sudo ln -s /usr/bin/python3 /usr/bin/python

################## BIOTOOLS ######################

mkdir -p $BIOTOOLS_PATH
cp ./install*.*sh $BIOTOOLS_PATH
cd $BIOTOOLS_PATH

bash install-nextflow.bash
bash install-fastqc.bash
bash install-fastp.bash
bash install-trimmomatic.bash
bash install-spades.bash
bash install-unicycler.bash
bash install-prokka.bash

printf "\n\nInstallation complete!\n\n"
printf "Apprentices, YOU ROCK!\n\n"


samples/pipeline-script.sh
#!/bin/bash

### Boilerplate

# Error handling
set -eo pipefail

### User Config

# Set global parameters (edit these parameters)
threads=4
baseDir=$PWD
read1=SRR11947553_sub_1.fq.gz
read2=SRR11947553_sub_2.fq.gz

### Setup

# Create new directory to store all results
mkdir $baseDir/bash_results
resultsDir=$baseDir/bash_results

# Create new directories for each task
mkdir $resultsDir/1.fastqc_results
mkdir $resultsDir/2.clean_reads
mkdir $resultsDir/3.genome_assembly
mkdir $resultsDir/4.annotate_genome

# Define paths to directories
fastqcDir=$resultsDir/1.fastqc_results
cleanDir=$resultsDir/2.clean_reads
assemblyDir=$resultsDir/3.genome_assembly
annotateDir=$resultsDir/4.annotate_genome

### Run Pipeline

# To run this script, navigate to the baseDir and type this command:
# bash run_bash.sh
# Note: you need to name this file run_bash.sh


# -------------------
# 1. Run FastQC and MultiQC
#--------------------

# Execute FastQC
fastqc --outdir $fastqcDir --threads ${threads} $baseDir/${read1} $baseDir/${read2} 

# Execute MultiQC
multiqc $fastqcDir --outdir $fastqcDir


# -------------------
# 2. Clean reads using fastp
#--------------------

# Execute fastp
fastp -i $baseDir/${read1} -I $baseDir/${read2} -o $cleanDir/${read1} -O $cleanDir/${read2} -j $cleanDir/fastp.json -h $cleanDir/fastp.html -q 30 --trim_poly_g --length_required 80 --thread ${threads}


# -------------------
# 3. Assemble clean reads using Unicycler
#--------------------

# Execute Unicycler
unicycler -1 $cleanDir/${read1} -2 $cleanDir/${read2} -o $assemblyDir --threads ${threads}


# -------------------
# 4. Annotate assembly using Prokka
#--------------------

# Execute Prokka
# conda activate prokka
prokka --outdir $annotateDir --force --prefix ${read1%_sub*} $assemblyDir/assembly.fasta


